upgrade/php/ps_900_migrate_category_images.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+function ps_900_migrate_category_images()
+{
+    /*
+     * Pre v9 worked like this:
+     * There could be 123.jpg, a cover image.
+     * There could be 123_thumb.jpg, a category miniature.
+     *
+     * If 123_thumb.jpg existed, it took it and generated category thumbnails from it.
+     * But, overwriting thumbnails of 123.jpg.
+     *
+     * So you suddenly got one thumbnail that was from a different image.
+     * Even worse, frontoffice and backoffice generated it as a different thumbnail,
+     * one as small, one as medium.
+     */
+    $formattedNameSmall = ImageType::getFormattedName('small');
+    $formattedNameMedium = ImageType::getFormattedName('medium');
+
+    // Get categories on shop
+    $categoryIds = Db::getInstance()->executeS('SELECT id_category FROM `' . _DB_PREFIX_ . 'category`');
```

Comment:
```suggestion
    $categoryIds = DbWrapper::executeS('SELECT id_category FROM `' . _DB_PREFIX_ . 'category`');
```

It has a new DbWrapper class which allows error handling during the update.

Reply:
@M0rgan01 Suggestion applied. :-)